onnxruntime/core/providers/cuda/shared_inc/fpgeneric.h

```diff
@@ -18,49 +18,175 @@
-  cublasSetMathMode(handle, CUBLAS_TENSOR_OP_MATH);
-  return cublasHgemmBatched(handle, transa, transb, m, n, k, alpha, (const __half**)Aarray, lda, (const __half**)Barray, ldb, beta, (__half**)Carray, ldc, batch_count);
+inline cublasStatus_t cublasGemmBatchedHelper(cublasHandle_t handle,
+                                              cublasOperation_t transa,
+                                              cublasOperation_t transb,
+                                              int m, int n, int k,
+                                              const float* alpha,
+                                              const float* Aarray[], int lda,
+                                              const float* Barray[], int ldb,
+                                              const float* beta,
+                                              float* Carray[], int ldc,
+                                              int batch_count,
+                                              const cudaDeviceProp& /*prop*/) {
+  return cublasSgemmBatched(handle,
+                            transa,
+                            transb,
+                            m, n, k,
+                            alpha,
+                            Aarray, lda,
+                            Barray, ldb,
+                            beta,
+                            Carray, ldc,
+                            batch_count);
+}
+inline cublasStatus_t cublasGemmBatchedHelper(cublasHandle_t handle,
+                                              cublasOperation_t transa,
+                                              cublasOperation_t transb,
+                                              int m, int n, int k,
+                                              const double* alpha,
+                                              const double* Aarray[], int lda,
+                                              const double* Barray[], int ldb,
+                                              const double* beta,
+                                              double* Carray[], int ldc,
+                                              int batch_count,
+                                              const cudaDeviceProp& /*prop*/) {
+  return cublasDgemmBatched(handle,
+                            transa,
+                            transb,
+                            m, n, k,
+                            alpha,
+                            Aarray, lda,
+                            Barray, ldb,
+                            beta,
+                            Carray, ldc,
+                            batch_count);
+}
+inline cublasStatus_t cublasGemmBatchedHelper(cublasHandle_t handle,
+                                              cublasOperation_t transa,
+                                              cublasOperation_t transb,
+                                              int m, int n, int k,
```

Comment:
>int m, int n, int k, [](start = 46, length = 20)

nit: some int parameters can be `const`.

Reply:
What I learned is that const for built-in types in function parameters with pass-by-value is not necessary. See this link: https://abseil.io/tips/109

Reply:
nice tip!
to quote from that link:
```
Do use top-level const on function parameters in definitions at your (or your team’s) discretion. You might follow the same rationale as you would for when to declare a function-local variable const.
```
i think const still has some usefulness here, similar to const local variables.

Reply:
I remember before when I use const to int as suggested here, the compiler would complain something like "it is not necessary to use const for built-in types.". Moreover, Nvidia doesn't use const for int in these cublas APIs either.

Reply:
The major benefit of `const` is for readability to me (so I marked my comment as nit).